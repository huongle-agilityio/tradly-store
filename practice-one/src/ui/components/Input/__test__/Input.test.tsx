import { Text } from 'react-native';
import { render, screen, fireEvent } from '@testing-library/react-native';

// Components
import { Input } from '..';

describe('Input Component', () => {
  it('Should match snapshot', () => {
    const { toJSON } = render(
      <Input value="Test" placeholder="Enter text" onChangeText={jest.fn()} />,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('Should render label if provided', () => {
    render(
      <Input
        label="Username"
        value=""
        placeholder="Enter text"
        onChangeText={jest.fn()}
      />,
    );

    expect(screen.getByText('Username')).toBeTruthy();
  });

  it('Should render icon if provided', () => {
    const MockIcon = () => <Text>🔍</Text>;

    render(
      <Input
        icon={<MockIcon />}
        value=""
        placeholder="Enter text"
        onChangeText={jest.fn()}
      />,
    );

    expect(screen.getByText('🔍')).toBeTruthy();
  });

  it('Should be disabled when disabled prop is true', () => {
    render(
      <Input
        value=""
        placeholder="Enter text"
        onChangeText={jest.fn()}
        disabled
      />,
    );

    const input = screen.getByPlaceholderText('Enter text');
    expect(input).toBeDisabled();
  });

  it('Should be disabled when isLoading prop is true', () => {
    render(
      <Input
        value=""
        placeholder="Enter text"
        onChangeText={jest.fn()}
        isLoading
      />,
    );

    const input = screen.getByPlaceholderText('Enter text');
    expect(input).toBeDisabled();
  });

  it('Should call onChangeText when input changes', () => {
    const mockOnChangeText = jest.fn();
    render(
      <Input
        value=""
        placeholder="Enter text"
        onChangeText={mockOnChangeText}
      />,
    );

    const input = screen.getByPlaceholderText('Enter text');
    fireEvent.changeText(input, 'Hello');

    expect(mockOnChangeText).toHaveBeenCalledWith('Hello');
  });
});
