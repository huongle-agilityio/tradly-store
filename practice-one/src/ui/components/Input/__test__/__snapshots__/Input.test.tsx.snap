// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component Should match snapshot 1`] = `
<View
  style={
    [
      {
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <View
    style={
      [
        {
          "borderColor": "#fff",
          "borderRadius": 24,
          "borderWidth": 1,
          "paddingHorizontal": 16,
          "paddingVertical": 13,
        },
        {
          "width": "100%",
        },
      ]
    }
  >
    <TextInput
      editable={true}
      onChangeText={[MockFunction]}
      placeholder="Enter text"
      placeholderTextColor="#fff"
      style={
        {
          "color": "#fff",
          "fontSize": 18,
          "lineHeight": 22,
        }
      }
      value="Test"
    />
  </View>
</View>
`;

exports[`Input Component Should render error message 1`] = `
<View
  style={
    [
      {
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <View
    style={
      [
        {
          "borderColor": "#fff",
          "borderRadius": 24,
          "borderWidth": 1,
          "paddingHorizontal": 16,
          "paddingVertical": 13,
        },
        {
          "width": "100%",
        },
      ]
    }
  >
    <TextInput
      editable={true}
      onChangeText={[MockFunction]}
      placeholder="Enter text"
      placeholderTextColor="#fff"
      style={
        {
          "color": "#fff",
          "fontSize": 18,
          "lineHeight": 22,
        }
      }
      value="Test"
    />
  </View>
  <Text
    style={
      [
        {
          "fontSize": 14,
          "lineHeight": 16,
        },
        {
          "fontWeight": "400",
        },
        {
          "color": "#901F2E",
        },
        {
          "paddingHorizontal": 10,
          "paddingTop": 12,
        },
      ]
    }
  >
    This field is required
  </Text>
</View>
`;
